### PR TITLE
Audit Log

### PR DESCRIPTION
**Pull request by jochman**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/32224

## Description
Get logs from EWS.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [x] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog_description.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Ews Audit Log
```
Comment (Itay4) [inline Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog_description.md:1]:
```suggestion
### Ews Audit Log
```

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog_description.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+This integration can search for logs.
+
+Supported authentication methods:
+
+- Basic authentication - Fill in the Email and password.
+- OAuth2.0 (For MFA enabled accounts) -
+    1. Enter a value for the UPN parameter in the integration configuration.
+    2. Run the ***ews-auth-start*** command and follow the instructions.
```
Comment (Itay4) [inline Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog_description.md:9]: update command names

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.ps1 [hunk 3]
```diff
@@ -0,0 +1,682 @@
...
+        # Creating ExchangeOnline client
+        $exo_client = [ExchangeOnlineClient]::new(
+                $integration_params.url, $integration_params.credentials.identifier,
+                $integration_params.credentials.password, $oauth2_client.access_token, $insecure, $no_proxy)
+        # Executing command
+        $demisto.Debug("Command being called is $command")
+        switch ($command)
+        {
+            "test-module" {
+                ($human_readable, $entry_context, $raw_response) = TestModuleCommand $oauth2_client $exo_client
```
Comment (Itay4) [inline Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.ps1:639]: verify test module works

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    - contextPath: O365.AuditLog.UserType
+      description: ''
+      type: Number
+    - contextPath: O365.AuditLog.Version
+      description: ''
+      type: Number
+    - contextPath: O365.AuditLog.Workload
+      description: ''
+      type: String
+    description: Tests the OAuth2.0 authorization process.
```
Comment (Itay4) [inline Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml:163]: update desc

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        in UTC. If empty, wll take current time.
+    - name: free_text
+      required: false
+      description: The free_text parameter filters the log entries by the specified
+        text string. If the value contains spaces, enclose the value in quotation
+        marks (").
+    - name: record_type
+      required: false
+      description: The record_type parameter filters the log entries by record type.
+        [Available record type](https://docs.microsoft.com/en-us/office/office-365-management-api/office-365-management-activity-api-schema#auditlogrecordtype)
```
Comment (Itay4) [inline Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml:55]: rm markdown

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+    - name: operations
+      required: false
+      isArray: true
+      description: The operations parameter filters the log entries by operation. The
+        available values for this parameter depend on the record_types value. [Available
+        values](https://docs.microsoft.com/en-us/microsoft-365/compliance/search-the-audit-log-in-security-and-compliance?view=o365-worldwide#audited-activities)
+    - name: user_ids
+      required: false
+      isArray: true
+      description: The UserIds parameter filters the log entries by the ID of the
```
Comment (Itay4) [inline Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml:70]: mention how to retrieve

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+  - name: o365-auditlog-auth-complete
+    arguments: []
+    description: Completes the OAuth2.0 authorization process.
+  - name: o365-auditlog-auth-test
+    arguments: []
+    description: Tests the OAuth2.0 authorization process.
+  - name: o365-search-auditlog
+    arguments:
+    - name: start_date
+      required: true
```
Comment (Itay4) [inline Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml:36]: make not mandatory and set default value

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.ps1 [hunk 8]
```diff
@@ -0,0 +1,682 @@
+$script:COMMAND_PREFIX = "o365"
+$script:INTEGRATION_ENTRY_CONTEXT = "O365"
```
Comment (Itay4) [inline Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.ps1:2]:
```suggestion
$script:INTEGRATION_ENTRY_CONTEXT = "O365AuditLog"
```

File: Packs/Office365AndAzureAuditLog/pack_metadata.json [hunk 9]
```diff
@@ -0,0 +1,15 @@
+{
+    "name": "Office365 and Azure Audit Log",
+    "description": "not now",
```
Comment (Itay4) [inline Packs/Office365AndAzureAuditLog/pack_metadata.json:3]: update description

**Comment by Itay4** [comment]
please add test playbook

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml [hunk 10]
```diff
@@ -0,0 +1,171 @@
+category: Data Enrichment & Threat Intelligence
```
Comment (Itay4) [inline Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml:1]: update category

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+        without a time zone, the value is in UTC.
+    - name: end_date
+      required: false
+      description: The end_date parameter specifies the end date of the date range.
+        Entries are stored in the unified audit log in Coordinated Universal Time
+        (UTC). If you specify a date/time value without a time zone, the value is
+        in UTC. If empty, wll take current time.
+    - name: free_text
+      required: false
+      description: The free_text parameter filters the log entries by the specified
```
Comment (Itay4) [inline Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml:49]: add example

File: Packs/Office365AndAzureAuditLog/pack_metadata.json [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+    "support": "xsoar",
+    "currentVersion": "1.0.0",
+    "author": "Cortex XSOAR",
+    "url": "https://www.paloaltonetworks.com/cortex",
+    "email": "",
+    "categories": [
+        "Data Enrichment & Threat Intelligence"
```
Comment (Itay4) [inline Packs/Office365AndAzureAuditLog/pack_metadata.json:10]: update category

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml (hunk 10 above)
Comment (jochman) [reply Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml:1]: This is the right category

File: Packs/Office365AndAzureAuditLog/pack_metadata.json (hunk 12 above)
Comment (jochman) [reply Packs/Office365AndAzureAuditLog/pack_metadata.json:10]: This is the right category

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.ps1 (hunk 3 above)
Comment (jochman) [reply Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.ps1:633]: Done

**Review by hod-alpert — APPROVED** [review]
* add this integration to unmockable

**Comment by ShirleyDenkberg** [comment]
@Itay4 @hod-alpert Doc review completed.

**Pull request merged**